es51x/src/main/scala/tech/beshu/ror/es/request/regular/RegularRequestHandler.scala

```diff
@@ -88,26 +93,58 @@ class RegularRequestHandler[Request <: ActionRequest, Response <: ActionResponse
   private def onAllow(requestContext: RequestContext,
                       requestInfo: RequestInfo,
                       blockContext: BlockContext): Unit = {
-    val searchListener = createSearchListener(requestContext, blockContext, engine.context)
-    requestInfo.writeResponseHeaders(BlockContextJavaHelper.responseHeadersFrom(blockContext).asJava)
-    requestInfo.writeToThreadContextHeaders(BlockContextJavaHelper.contextHeadersFrom(blockContext).asJava)
-    requestInfo.writeIndices(BlockContextJavaHelper.indicesFrom(blockContext).asJava)
-    requestInfo.writeSnapshots(BlockContextJavaHelper.snapshotsFrom(blockContext).asJava)
-    requestInfo.writeRepositories(BlockContextJavaHelper.repositoriesFrom(blockContext).asJava)
+    requestContext.uriPath match {
+      case CatIndicesPath(_) if emptySetOfFoundIndices(blockContext) =>
+        baseListener.onResponse(emptyClusterStateResponse)
+      case CatTemplatePath(_) | TemplatePath(_) =>
+        val searchListener = createSearchListener(requestContext, blockContext, engine.context)
+        applyIfExists(indicesFrom(blockContext)) {
+          requestInfo.writeTemplatesOf
+        }
+        writeCommonParts(requestInfo, blockContext)
+        proceed(searchListener)
+      case _ =>
+        val searchListener = createSearchListener(requestContext, blockContext, engine.context)
+        applyIfExists(indicesFrom(blockContext)) {
+          requestInfo.writeIndices
+        }
+        requestInfo.writeSnapshots(BlockContextRawDataHelper.snapshotsFrom(blockContext))
+        requestInfo.writeRepositories(BlockContextRawDataHelper.repositoriesFrom(blockContext))
+        writeCommonParts(requestInfo, blockContext)
+
+        proceed(searchListener)
+    }
+  }
 
-    proceed(searchListener)
+  private def applyIfExists[T](outcome: Outcome[T])(function: T => Unit): Unit = {
+    outcome match {
+      case Outcome.Exist(value) => function(value)
+      case Outcome.NotExist =>
+    }
+  }
+
+  private def writeCommonParts(requestInfo: RequestInfo, blockContext: BlockContext): Unit = {
+    requestInfo.writeResponseHeaders(BlockContextRawDataHelper.responseHeadersFrom(blockContext))
+    requestInfo.writeToThreadContextHeaders(BlockContextRawDataHelper.contextHeadersFrom(blockContext))
+  }
+
+  private def emptySetOfFoundIndices(blockContext: BlockContext) = {
```

Comment:
this too looks's like doubled `Option.forall(_.isEmpty)`, or something like tkis